zilliqa/src/block_store.rs

```diff
@@ -50,12 +52,19 @@ pub struct BlockStore {
     message_sender: MessageSender,
 }
 
+#[derive(Clone, Debug, Default)]
+struct PeerInfo {
+    highest_known_view: u64,
+    requested_blocks: u64,
```

Comment:
I bet this is being reworked, so feel free to ignore me, but would be good to have comments here to say what gets stored in these (and how reliable it needs to be)

If it were me, I'd abstract this out - have a block_fetcher.rs and make peers a block_fetcher::Context - that makes it rather clearer where the logic is, and gives you more contained diffs when you inevitably make the data structure more complex.

Eventually,  you also want to age out this data so that misconceptions and errors correct themselves over time (in particular, eventually - actually quite soon, if Theo's PR is to be believed - highest_known_view will need to be joined by lowest_known_view, or more likely a Range .. )

Reply:
My next indeed changes this so I can retry failed requests - I will add extra details there.